lib/link_checker/uri_checker/http_checker.rb

```diff
@@ -284,10 +284,22 @@ def gov_uk_uri?
       Plek.new.website_root.include?(uri.host)
     end
 
+    def rate_limit_header
+      return {} unless gov_uk_uri?
+      { "Rate-Limit-Token": Rails.application.secrets.rate_limit_token }
+    end
+
+    def basic_authorization_header
+      return {} unless LinkCheckerApi.authed_domains.include?(uri.host)
+      { "Authorization": LinkCheckerApi.authed_domains[uri.host] }
+    end
+
     def additional_connection_headers
-      return nil unless gov_uk_uri?
+      headers = {}
```

Comment:
We could possibly make this more succinct with:

```ruby
def ...
  Hash.new
    .merge(rate_limit_header)
    .merge(basic_authorization_header)
end
```

Reply:
YES! @lukemorton and I were looking at this and thinking "how can we make this prettier"